tests/byzantium/eip198_modexp_precompile/test_modexp.py

```diff
@@ -0,0 +1,191 @@
+                "0000000000000000000000000000000000000000000000000000000000000020"
+                "03"
+                "ffff"
+                "80"
+            ],
+            ["0x01", "0x3b01b01ac41f2d6e917c6d6a221ce793802469026d9ab7578fa2e79e4da6aaab"],
+            id="EIP-198-case5",
+        ),
+    ],
+)
+def test_modexp(state_test: StateTestFiller, input: List[str], output: str):
+    """
+    Test the MODEXP precompile
+    """
+    env = Environment()
+    pre = {TestAddress: Account(balance=1000000000000000000000)}
+    post = {}
+
+    account = to_address(0x100)
+
+    pre[account] = Account(
+        code=(
+            # Store all CALLDATA into memory (offset 0)
+            Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE())
+            # Store the returned CALL status (success = 1, fail = 0) into slot 0:
+            + Op.SSTORE(
+                0,
+                # Setup stack to CALL into ModExp with the CALLDATA and CALL into it (+ pop value)
+                Op.CALL(Op.GAS(), 0x05, 0, 0, Op.CALLDATASIZE(), 0, 0),
+            )
+            # Store contract deployment code to deploy the returned data from ModExp as
+            # contract code (16 bytes)
+            + Op.MSTORE(
+                0,
+                (
+                    (
+                        # Need to `ljust` this PUSH32 in order to ensure the code starts
+                        # in memory at offset 0 (memory right-aligns stack items which are not
+                        # 32 bytes)
+                        Op.PUSH32(
+                            (
+                                Op.CODECOPY(0, 16, Op.SUB(Op.CODESIZE(), 16))
+                                + Op.RETURN(0, Op.SUB(Op.CODESIZE, 16))
+                            ).ljust(32, bytes(1))
+                        )
+                    )
+                ),
+            )
+            # RETURNDATACOPY the returned data from ModExp into memory (offset 16 bytes)
+            + Op.RETURNDATACOPY(16, 0, Op.RETURNDATASIZE())
+            # CREATE contract with the deployment code + the returned data from ModExp
+            + Op.CREATE(0, 0, Op.ADD(16, Op.RETURNDATASIZE()))
+            # STOP (handy for tracing)
+            + Op.STOP()
+        )
+    )
+
+    data = ""
+    if len(input) == 1:
+        data = input[0]
+    else:
+        if len(input) < 4:
+            input.append("")
+        data = create_modexp_tx_data(*input)
+
+    tx = Transaction(
+        ty=0x0,
+        nonce=0,
+        to=account,
+        data=data,
+        gas_limit=500000,
+        gas_price=10,
+        protected=True,
+    )
+    if output[0] != "0x00":
+        # Note: This account is only created if the CALL output is 1. Otherwise, it is not created.
+        contract_address = compute_create_address(account, tx.nonce)
+        post[contract_address] = Account(code=output[1])
+    post[account] = Account(storage=dict([("0x00", output[0])]))
```

Comment:
Just wanted to point out this is a more compact way of defining a dict with one value:
```suggestion
    post[account] = Account(storage={"0x00": output[0]})
```
No need to apply this, I modified it in my PRs (https://github.com/jochem-brouwer/execution-spec-tests/pull/2 and https://github.com/jochem-brouwer/execution-spec-tests/pull/3).